src/common/CMakeLists.txt

```diff
@@ -225,6 +225,7 @@ elseif(HAVE_POWER8)
     list(APPEND crc32_srcs
       crc32c_ppc_asm.S
       crc32c_ppc_fast_zero_asm.S)
+    set_source_files_properties(crc32c_ppc_asm.S PROPERTIES COMPILE_FLAGS -D__ASSEMBLY__)
```

Comment:
why does this definition help to get these symbols visible?

Reply:
...and how does it match Ken's comments that main builds on power ok, but squid must be missing something from main?

Reply:
and why doesn't anything in your PR or commit message refer to the fact that this is squid-specific, or clarify it in the bug?

Reply:
> why does this definition help to get these symbols visible?
According to src/common/crc32c_ppc_constants.h, these symbols are included only if __ASSEMBLY__ is defined.

Reply:
> ...and how does it match Ken's comments that main builds on power ok, but squid must be missing something from main?
>and why doesn't anything in your PR or commit message refer to the fact that this is squid-specific, or clarify it in the bug?

When I tested, build is failing for both the main and squid branches with the same error.

Reply:
> > why does this definition help to get these symbols visible?
> 
> According to src/common/crc32c_ppc_constants.h, these symbols are included only if `__ASSEMBLY__` is defined.

@sudeeshjohn  thank you. could you update the commit message to note this. as "Fixes build failure on IBM Power" is not accurate enough, and it does not summarize the change made in this commit.

Reply:
@tchaikov just updated the commit message, please take a look.

Reply:
much better now, but could you use a shorter title, and put the details in the body of the commit message? something like:

```
build: define __ASSEMBLY__ when building crc32c_ppc_asm.S

/// more details go here

```

Reply:
sorry for adding such a long title, updated as you suggested